src/utils/validateProps.ts

```
@@ -1,36 +1,69 @@
 import errors from '../dictionary/errors.json';
-import { PrivateKey } from '@hashgraph/sdk';
-
-export const validateProps = ({
-  batchSize = null,
-  tokenId = null,
-  amount = null,
-  metaData = null,
-  supplyKey = null,
-  pathToMetadataURIsFile = null,
-  metadataArray = null,
-}: {
-  batchSize?: number | null;
-  tokenId?: string | null;
-  amount?: number | null;
-  metaData?: string | null;
-  supplyKey?: PrivateKey | null;
-  pathToMetadataURIsFile?: string | null;
-  metadataArray?: string[] | null;
-}) => {
-  if (batchSize !== null && batchSize > 10) throw new Error(errors.maxBatchSize);
-  if (batchSize !== null && batchSize < 1) throw new Error(errors.minBatchSize);
-  if (tokenId !== null && !tokenId) throw new Error(errors.tokenIdRequired);
-  if (amount !== null && (!amount || amount < 1)) throw new Error(errors.minAmount);
-  if (metaData !== null && !metaData) throw new Error(errors.metadataRequired);
-  if (supplyKey !== null && !supplyKey) throw new Error(errors.supplyKeyRequired);
-  if (pathToMetadataURIsFile !== null && !pathToMetadataURIsFile)
-    throw new Error(errors.pathRequired);
-  if (
-    metadataArray !== null &&
-    pathToMetadataURIsFile !== null &&
-    !metadataArray &&
-    !pathToMetadataURIsFile
-  )
-    throw new Error(errors.csvOrArrayRequired);
+import { PropsType } from '../types/validateProps';
+
+export const validateProps = (props: PropsType) => {
+  validateSupplyKey(props);
+  validateBatchSize(props);
+  validateTokenId(props);
+  validateAmount(props);
+  validateMetaData(props);
+  validateMetadataArray(props);
+  validatePathToMetadataURIsFile(props);
+  validateIsUnique(props);
+};
+
+const validateSupplyKey = (props: PropsType) => {
+  if (Object.prototype.hasOwnProperty.call(props, 'supplyKey')) {
```

Why do you need this line? Cannot you just check it like this: 
`  if (supplyKey !== null && !supplyKey) throw new Error(errors.supplyKeyRequired);`

---

Nope. I want this function as generic as it can be. 

The Object.prototype.hasOwnProperty.call(props, 'batchSize') is used to check if the props object has a property named batchSize. This method will return true even if the value of the property is null or undefined.  

On the other hand, props.batchSize !== null only checks if the batchSize property is not null. It doesn't check if the props object actually has a batchSize property. Also, it will return true if batchSize is undefined.  

So, Object.prototype.hasOwnProperty.call(props, 'batchSize') is a more thorough check because it ensures that the batchSize property exists on the props object, regardless of its value.

For example without this line, this test would be successful which is not supposed to:

  it('should throw an error if supplyKey is not provided', () => {
    expect(() => validateProps({ supplyKey: undefined })).toThrow(errors.supplyKeyRequired);
  });